pkg/pillar/cmd/zedrouter/server.go

```diff
@@ -227,3 +253,238 @@ func (hdl hostnameHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.Write(resp)
 	}
 }
+
+// ServeHTTP for metadataHandler process requests for ec2-compatible requests
+func (hdl metadataHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Errorf("ServeHTTP metadataHandler %s", r.URL.Path)
+	dirname, filename := path.Split(strings.TrimSuffix(r.URL.Path, "/"))
+	dirname = strings.TrimSuffix(dirname, "/")
+	remoteIP := net.ParseIP(strings.Split(r.RemoteAddr, ":")[0])
+	var hostname string
+	var id string
+	anStatus := lookupAppNetworkStatusByAppIP(hdl.ctx, remoteIP)
+	if anStatus != nil {
+		hostname = anStatus.DisplayName
+		id = anStatus.UUIDandVersion.UUID.String()
+	} else {
+		log.Errorf("No AppNetworkStatus for %s",
+			remoteIP.String())
+		w.WriteHeader(http.StatusNotFound)
```

Comment:
The lookup which failed was on the source IP, which indicates that the server hasn't (yet) determined the IP address of the app instance. Note an unknown URL from the client.
Thus I think a 503 Service Unavailable is more appropriate here, and I think the same logic applies for the cases where you return internal server error.